internals/secrethub/variable_reader_test.go

```diff
@@ -0,0 +1,144 @@
+				"DIFFERENT_PREFIX_TEST":          "other_test_value",
+				templateVarEnvVarPrefix + "TEST": "test_value",
+				"TEST":                           "yet_another_test_value",
+			},
+			commandTemplateVars: nil,
+			constructorErr:      nil,
+			variableToRead:      "test",
+			expectedValue:       "test_value",
+			readErr:             nil,
+		},
+		"command_template_vars_success": {
+			osEnv: nil,
+			commandTemplateVars: map[string]string{
+				templateVarEnvVarPrefix + "TEST": "test_value",
+				"DIFFERENT_PREFIX_TEST":          "other_test_value",
+				"TEST":                           "yet_another_test_value",
+			},
+			constructorErr: nil,
+			variableToRead: "test",
+			expectedValue:  "yet_another_test_value",
+			readErr:        nil,
+		},
+		"variable_not_existent": {
+			osEnv: map[string]string{
+				templateVarEnvVarPrefix + "TEST1": "testA",
+				templateVarEnvVarPrefix + "TEST2": "testB",
+			},
+			commandTemplateVars: map[string]string{
+				"test3": "testC",
+				"test4": "testD",
+			},
+			constructorErr: nil,
+			variableToRead: "test5",
+			expectedValue:  "",
+			readErr:        tpl.ErrTemplateVarNotFound("test5"),
+		},
+		"os_var_name_not_posix": {
+			osEnv: map[string]string{
+				templateVarEnvVarPrefix + "TEST-1": "testA",
+				templateVarEnvVarPrefix + "TEST2":  "testB",
+			},
+			commandTemplateVars: map[string]string{
+				"test3": "testC",
+				"test4": "testD",
+			},
+			constructorErr: ErrInvalidTemplateVar("test-1"),
+			variableToRead: "",
+			expectedValue:  "",
+			readErr:        nil,
+		},
+		"command_var_name_not_posix": {
+			osEnv: map[string]string{
+				templateVarEnvVarPrefix + "TEST1": "testA",
+				templateVarEnvVarPrefix + "TEST2": "testB",
+			},
+			commandTemplateVars: map[string]string{
+				"3test3": "testC",
+				"test4":  "testD",
+			},
+			constructorErr: ErrInvalidTemplateVar("3test3"),
+			variableToRead: "",
+			expectedValue:  "",
+			readErr:        nil,
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			reader, err := newVariableReader(tc.osEnv, tc.commandTemplateVars)
+			if err != nil {
+				assert.Equal(t, err, tc.constructorErr)
+				return
+			}
+
+			value, err := reader.ReadVariable(tc.variableToRead)
+			if err != nil {
+				assert.Equal(t, err, tc.readErr)
+			}
+
+			assert.Equal(t, value, tc.expectedValue)
+		})
+	}
+}
+
+func TestPromptVariableReader(t *testing.T) {
+	osEnv := map[string]string{
+		templateVarEnvVarPrefix + "TEST1": "testA",
+		templateVarEnvVarPrefix + "TEST2": "testB",
+	}
+	commandTemplateVars := map[string]string{
+		"test3": "testC",
+		"test4": "testD",
+	}
+
+	reader, err := newVariableReader(osEnv, commandTemplateVars)
+	assert.OK(t, err)
+	io := ui.NewFakeIO()
+	io.PromptIn.Reads = []string{"foobar\n"}
+	reader = newPromptMissingVariableReader(reader, io)
+
+	val, err := reader.ReadVariable("test5")
+	assert.Equal(t, val, "foobar")
+}
+
+func TestPromptVariableReader_NoPrompt(t *testing.T) {
```

Comment:
You could combine these two prompt variable reader testcases to share their setup (as it is the same):

```Go
func TestPromptVariableReader(t *testing.T) {

  // setup code

  t.Run("prompt", func(t *testing.T) {
    // prompt testcase
  })

  t.Run("no prompt", func(t *testing.T) {
    // no prompt testcase
  })
}
```

Reply:
Good idea!